### PR TITLE
Test for banned imports.

### PR DESCRIPTION
**Pull request by n3wscott**
/assign @grantr

### CODE REVIEW

File: test/banned_deps_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	dir := filepath.Dir(filename)
+	dir = fmt.Sprintf("%s%s..%s%s", dir, string(filepath.Separator), string(filepath.Separator), "vendor")
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			return nil
+		}
+		for _, banned := range bannedImports {
+			if strings.HasSuffix(path, banned) {
+				return fmt.Errorf("%s is a banned import", banned)
```
Comment (grantr) [inline test/banned_deps_test.go]: Expand this to be more clear that there's an easy alternative. `%s is a banned import. Use the vanity import path instead.`

File: test/banned_deps_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	}, {
+		banned: "github.com/kubernetes",
+		vanity: "k8s.io",
+	}, {
+		banned: "github.com/istio",
+		vanity: "istio.io",
+	},
+}
+
+func TestBannedImports(t *testing.T) {
```
Comment (matzew) [inline test/banned_deps_test.go]: could we use this as "helpers", from eventing, like here:

https://github.com/knative/eventing-contrib/blob/master/test/e2e/channel_chain_test.go#L28

**Comment by n3wscott** [comment]
/hold

trying to upstream to pkg

**Comment by matzew** [comment]
what's the status here ?

**Comment by antoineco** [comment]
/close
ref. knative/pkg#696

**Pull request closed without merging**